Update README.md (#54)
IllusionMods/HS2-Translation: README.md

# HS2 Translation

English fan translation project for Honey Select 2. The translations are applied while the game is running and do not require replacing or modifying any game files.

## Prerequisites

- [BepInEx 5.4](https://github.com/BepInEx/BepInEx/releases/tag/v5.4.8)
- [BepisPlugins for HS2](https://github.com/bbepis/BepisPlugins/releases)
- [XUnity.AutoTranslator](https://github.com/bbepis/XUnity.AutoTranslator)
- [HS2_TextResourceRedirector](https://github.com/IllusionMods/TranslationTools#textresourceredirector) (required for most resources)
- [HS2_Subtitles](https://github.com/DeathWeasel1337/KK_Plugins#subtitles) (required to see subtitles)

> - [HS2_TranslationHelper.v1.1](https://github.com/GeBo1/GeBoPlugins/releases/tag/r16) (Optional, but recommended. Translates Japanese names into the Western alphabet without trying to translate into English avoiding mistakes like 悟飯=Food instead of 悟飯=Gohan).

Reviewer: I'm ok with listing `TranslationHelper` here, but I'd like to clarify the description a bit, because while it does sort of do that as a result of what it actually does, it's a bit misleading.

Also I'd like to leave `TranslationCacheCleaner` mentioned in here (see below)

Author: Hmm, I'll let the explanation of TranslationHelper for you, because I really don't understand it...
Removing the TranslationCacheCleaner was an oopsie from my part, will add again.

## Installation

1. Ensure you have the prerequisites installed.

> 2. Delete the "**en**" folder in BepInExEx\Translation\\**en** (if it exists)

Reviewer: I do **not** like telling them to delete this folder for numerous reasons. I understand why it's being suggested, but this fan translation doesn't own that folder. Other plugins/mods/etc. may add files here than can safely co-exist.  I realize you make all-in-one packages, and in that case this would make sense.  Maybe 2 sets of install instructions, one of which has the delete with a big fat warning?

Author: For now I'll just remove this part. I'll think something later. Maybe list the folders to be deleted?

Any help is appreciated. Regardless of your translation skill and Japanese knowledge you can still help with translations. Even if you have no experience, you can help by proofreading or using machine translation services such as Google Translate or [DeepL](https://www.deepl.com/translator) (we currently recommend to use [DeepL](https://www.deepl.com/translator)). In the case of machine translations, clean up the translation using sanity and a little logic.

Translations are all inside of `Bepinex\Translation\en\` folder. They are then split into the following folders:
- `RedirectedResources` - Replacements for strings embedded in asset files. Preferred over `Text` for accuracy and performance.
- `Text` - Normal text replacements and modifications.
- `Text\Localizations` - Strings under this folder were dumped via TextDump. Translations can be added for missing entries, but new entries should not be added or merging future dumps will become difficult.

> - `Text\zz_MachineTranslations` - All raw machine translations (like google translate) go here. Translated/cleaned up entries must be correctly sent to other folders. The goal is to proper translate until this folder becomes empty.

Reviewer: Maybe leave mention of `Text\zz_MachineTranslations` until we're 100% sure we won't use it again for something in the near future.

Author: Will leave a mention about being the old place for MTL

> Translation done purely by machines should be kept in the designated files (`zz_machineTranslation.txt` files within `RedirectedResources`), but the goal of this project is that none of these files are left and everything is translated properly.

Reviewer: ```suggestion
Translation done purely by machines must be kept in the designated files (`zz_machineTranslation.txt` files within `RedirectedResources`). These should be considered placeholders until manual translations are available (those should go into `translation.txt`). The goal is to have everything, manually translated.
```

> Each translation line follows the pattern: `original text=translated text`. Example: `悟飯=Food`.

Reviewer: This would be easier to read using code blocks (which is hard to insert as an example here).

after "pattern:" newline, 3 back-quotes, newline, your example, newline, 3 back-quotes, newline, "Example:", newline, 3 back-quotes, newline, your example, newline, 3 back-quotes, newline, 

Output would look like this:

> Each translation line follows the pattern: 
> ```
> original text=translated text
> ```
> Example: 
> ```
> 悟飯=Food
> ```

Author: Shamelessly copied this code and pasted to the file :P

> - `Texture` - Contains the translated version of game images. The game images are translated using Photoshop, and the .psd files can be found in the GitHub's folder `Image Source Files`.

Reviewer: ```suggestion
- `Texture` - Contains the translated version of game images. The game images are translated using image editing software. If available and the source files (`.psd`, `.xcf`, etc.) files can be found in the proejcts `Image Source Files` folder.
```

Coordinate with other translators on the [Illusion Soft Discord server](https://discord.gg/illusionsoft) #translation channel. To avoid translation conflicts please ask if anyone is working on a file. If you have any questions about the quality of your translations, ask for advice on the server.

### How to add or improve translations

- If you want to make a simple edit simply open the file in question and click edit. After you are done editing, commit the changes and stall a pull request.
- If you have more translations to submit [fork the repository](https://help.github.com/articles/fork-a-repo/). This will make a copy of the original project in your account. Upload your changes to the fork into your account, and then [send a pull request](https://help.github.com/articles/about-pull-requests/) to the original project. Your pull request will be reviewed and accepted after a quality check. Again, avoid raw machine translations. Proper capitalization, punctuation, and spelling is a must.

> ## Translating the "Text" folder

Reviewer: Please revert this. It's linked to elsewhere and if you change the headings the links break

```suggestion
## Text Translations
```

Texts that are not translated in `RedirectedResources` use the `Text` directory. In this directory the first translation found for a given text is used every time that this same text is found. But sometimes the same word can have different meanings depending on where you are in the game, in this case you can use the "scope level" feature to tell which translation should be used in that part of the game.

### Scope Levels

To use scope levels, use the following structure:
```
#set level xxx

> text=translation

Reviewer: Maybe give example with two entries inside the scope so people don't think you have to scope each line individually?

#unset level xxx
```
Where "xxx" represents the number of the desired scope level. Below is a table containing the known scope levels and their locations:

> ### Known Scope Levels

Reviewer: Make this a sub-section of above
```suggestion
#### Known Scope Levels
```

| 14    | Character Search      |
| 15    | Vip Lobby             |

> ## Translating the "RedirectedResources" folder

Reviewer: ```suggestion
##  Resource Translation
```

Add the folder name info below instead.

This is the main translation directory for the game, so it gets special treatment.

> Inside each folder there is a file called "translation.txt", this is the file where the translations should go. In some folders there is also the file "zz_machineTranslation.txt" which is where machine translations go, that usually have a lower quality.

Reviewer: ```suggestion
Inside each folder there is a file named `translation.txt`, this is the file where the translations should go. In some folders there is also the file `zz_machineTranslation.txt` which is where machine translations go, that usually have a lower quality.
```

> Every translation.txt file has the raw Japanese text that needs translations. Each one starts commented out (`//` at the begining) so it will not be loaded. For your text to display correctly in game, put the translation on the right side of the equal sign and remove the `//` at the start of the line. Do not edit the Japanese text or the translation will not work. Example:

Reviewer: ```suggestion
Every `translation.txt` file has the raw Japanese text that needs translations. Each one starts commented out (`//` at the begining) so it will not be loaded. For your text to display correctly in game, put the translation on the right side of the equal sign and remove the `//` at the start of the line. Do not edit the Japanese text or the translation will not work. Example:
```

Before:
```
//Tシャツ=
```
After:
```
Tシャツ=T-shirt
```

> The file "zz_machineTranslation.txt" can only have the translation lines that are commented out in "translation.txt". If the file "translation.txt" has been fully translated, the file "zz_machineTranslation.txt" should be deleted. The goal of this translation project is that none of the "zz_machineTranslation.txt" files remains.

Reviewer: None of this is technically true, and I find it confusing.  I don't really think I want people removing lines in PRs from `zz_machineTranslation.txt` so we don't accidentally have later changes during review cause us to drop lines. I'd rather request people comment the lines out (although leaving them won't hurt since they won't get hit)

```suggestion
The file `zz_machineTranslation.txt` can only have the translation lines that are commented out in `translation.txt`. If the file `translation.txt` has been fully translated, the file `zz_machineTranslation.txt` should be deleted. The goal of this translation project is that none of the `zz_machineTranslation.txt`" files remain.
```
